### PR TITLE
[Hub Generated] Review request for Microsoft.Authorization to add version preview/2020-04-01-preview

### PR DESCRIPTION
**Pull request by dagoroz**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/dagoroz/azure-rest-api-specs/dev-authorization-Microsoft.Authorization-2020-04-01-preview...master/).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for AME onboarding.
- [x] [Validation tools](https://aka.ms/swaggertools) have been run on swagger spec(s) updated and errors have all been fixed with this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, [please view the FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

- [x] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories.
*	adding/removing APIs.
* adding/removing properties.
* adding/removing API-version.
* adding a new service in Azure.

*Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.*

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them.

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by dagoroz** [comment]
@yungezz gentle ping for review

File: specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-04-01-preview/authorization-RoleAssignmentsCalls.json [hunk 1]
```diff
@@ -0,0 +1,687 @@
...
+        },
+        "canDelegate": {
+          "type": "boolean",
+          "description": "The Delegation flag for the role assignment"
+        },
+        "description": {
+          "type": "string",
+          "description": "Description of role assignment"
+        },
+        "condition": {
```
Comment (pilor) [inline specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-04-01-preview/authorization-RoleAssignmentsCalls.json:536]: condition and conditionVersion need better descriptions that describe what the values can be and what the properties do.  There also need to be examples added that use these new properties.

**Review by pilor — CHANGES_REQUESTED** [review]

**Comment by yungezz** [comment]
hi @dagoroz, the change looks good to me. Pls help to address ARM board comments. The PR is ok to merge after ARM signoff.

File: specification/authorization/resource-manager/readme.md [hunk 2]
```diff
@@ -4,39 +4,83 @@
...
-
-
 ### Basic Information
+
 These are the global settings for the Authorization API.
 
 ``` yaml
 openapi-type: arm
-tag: package-2018-09-01-preview
+tag: package-preview-2020-04
```
Comment (yungezz) [inline specification/authorization/resource-manager/readme.md:29]: pls use the api version `2020-04-preview`

File: specification/authorization/resource-manager/readme.md [hunk 3]
```diff
@@ -4,39 +4,83 @@
...
+    where: $.definitions.DenyAssignmentProperties.properties.doNotApplyToChildScopes
+    from: authorization-DenyAssignmentGetCalls.json
+    reason: for this case the result of the proposed change would resemble a boolean anyways
+  - suppress: EnumInsteadOfBoolean
+    where: $.definitions.DenyAssignmentProperties.properties.isSystemProtected
+    from: authorization-DenyAssignmentGetCalls.json
+    reason: for this case the result of the proposed change would resemble a boolean anyways
+```
+
+### Tag: package-preview-2020-04
```
Comment (yungezz) [inline specification/authorization/resource-manager/readme.md:72]: same as  above

**Comment by yungezz** [comment]
hi @dagoroz could you pls look at review comments?

File: specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-04-01-preview/authorization-RoleAssignmentsCalls.json (hunk 1 above)
Comment (dagoroz) [reply specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-04-01-preview/authorization-RoleAssignmentsCalls.json:536]: added

File: specification/authorization/resource-manager/readme.md (hunk 2 above)
Comment (dagoroz) [reply specification/authorization/resource-manager/readme.md]: fixed

**Comment by dagoroz** [comment]
@yungezz @pilor all the comments have been addressed
please approve

**Review by pilor — APPROVED** [review]

**Review by yungezz — APPROVED** [review]

**Pull request merged**